walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQueryBogus.java

```diff
@@ -0,0 +1,131 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Hashtable;
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public final class FindMeetingQueryBogus {
+  public Boolean areAttendeesIncluded(Collection<Event> events, MeetingRequest request){
+      HashSet<String> bookedEventAttendees = new HashSet<>();
+      for (Event event: events) {
+          bookedEventAttendees.addAll(event.getAttendees());
+      }
+
+      for(String xx : bookedEventAttendees) {
+          System.out.println(xx);
+      }
+
+      bookedEventAttendees.retainAll(request.getAttendees());
+      System.out.println(bookedEventAttendees.isEmpty());
+      return !bookedEventAttendees.isEmpty();
+  }
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    long whole_day = 24*60;
+
+    if (request.getAttendees().isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    else if (request.getDuration() > whole_day) {
+        return Arrays.asList();
+    }
+    else if (!areAttendeesIncluded(events, request)) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    System.out.println(!areAttendeesIncluded(events, request));
+
+    ArrayList<Integer[]> isAttendeeBooked = new ArrayList<Integer[]>(); 
+
+    Collection<TimeRange> time_result = new ArrayList<TimeRange>();
+    for(Event event: events) {
+        Integer start = event.getWhen().start();
+        Integer end = event.getWhen().end();
+        Integer duration = event.getWhen().duration();
+        System.out.println(start + " " + end + " -- " + duration);
+        Integer[] attendeeBusyTimeRange = new Integer[3];
+        if (isAttendeeBooked.isEmpty()) {
+            attendeeBusyTimeRange[0] = start;
+            attendeeBusyTimeRange[1] = duration;
+            attendeeBusyTimeRange[2] = end;
+            isAttendeeBooked.add(attendeeBusyTimeRange);
+            continue;
+        }
+        int length = isAttendeeBooked.size();
+        for (int i =0 ; i < length; i++) {
+            Integer[] attendeeTime = isAttendeeBooked.get(i);
+            if (Integer.compare(start, attendeeTime[2]) > 0) {
+                attendeeBusyTimeRange[0] = start;
+                attendeeBusyTimeRange[1] = duration;
+                attendeeBusyTimeRange[2] = end;
+                isAttendeeBooked.add(attendeeBusyTimeRange);   
+            }
+            else if(Integer.compare(start, attendeeTime[2]) <=0){
+                if(Integer.compare(end, attendeeTime[2]) < 0){
```

Comment:
deeply nested if statements are discouraged. I think this if statement can be combined with the one above.

Reply:
All changes have been reviewed and modified.